editor/src/components/canvas/canvas-strategies/draw-to-insert-strategy.spec.browser2.tsx

```diff
@@ -0,0 +1,766 @@
+              position: 'absolute',
+              left: 10,
+              top: 10,
+              width: 380,
+              height: 180,
+              backgroundColor: '#d3d3d3',
+            }}
+          >
+            <div
+              data-uid='ddd'
+              style={{
+                position: 'absolute',
+                left: 5,
+                top: 5,
+                width: 1000,
+                height: 10,
+              }}
+            />
+          </div>
+          <div
+            data-uid='ccc'
+            style={{
+              position: 'absolute',
+              left: 10,
+              top: 200,
+              width: 380,
+              height: 190,
+              backgroundColor: '#FF0000',
+            }}
+          />
+        </div>
+      `),
+    )
+  })
+})
+
+describe('Inserting into flex row', () => {
+  before(() => {
+    viewport.set(2200, 1000)
+  })
+
+  const newElementUID = 'ddd'
+  const newElement = jsxElement(
+    'div',
+    newElementUID,
+    [
+      simpleAttribute('data-uid', newElementUID),
+      simpleAttribute('style', { position: 'absolute' }),
+    ],
+    [],
+  )
+
+  async function startInsertMode(
+    dispatch: (actions: ReadonlyArray<EditorAction>, waitForDOMReport: boolean) => Promise<void>,
+  ) {
+    await act(() =>
+      dispatch([enableInsertModeForJSXElement(newElement, newElementUID, {}, null)], false),
+    )
+  }
+
+  const inputCode = makeTestProjectCodeWithSnippet(`
+    <div
+      data-uid='aaa'
+      style={{
+        width: '100%',
+        height: '100%',
+        backgroundColor: '#FFFFFF',
+        position: 'relative',
+        display: 'flex',
+        gap: 10,
+      }}
+    >
+      <div
+        data-uid='bbb'
+        data-testid='bbb'
+        style={{
+          width: 180,
+          height: 180,
+          backgroundColor: '#d3d3d3',
+        }}
+      />
+      <div
+        data-uid='ccc'
+        style={{
+          width: 100,
+          height: 190,
+          backgroundColor: '#FF0000',
+        }}
+      />
+    </div>
+  `)
+
+  it('Insert into zero position in flex', async () => {
+    const renderResult = await renderTestEditorWithCode(inputCode, 'await-first-dom-report')
+    await startInsertMode(renderResult.dispatch)
+
+    const targetElement = renderResult.renderedDOM.getByTestId('bbb')
+    const targetElementBounds = targetElement.getBoundingClientRect()
+    const canvasControlsLayer = renderResult.renderedDOM.getByTestId(CanvasControlsContainerID)
+
+    const startPoint = slightlyOffsetWindowPointBecauseVeryWeirdIssue({
+      x: targetElementBounds.x + 5,
+      y: targetElementBounds.y + 5,
+    })
+    const endPoint = slightlyOffsetWindowPointBecauseVeryWeirdIssue({
+      x: targetElementBounds.x + 25,
+      y: targetElementBounds.y + 305,
+    })
+
+    // Drag from inside bbb to inside ccc
+    await fireDragEvent(canvasControlsLayer, startPoint, endPoint)
+
+    // Check that the inserted element is a child of bbb
```

Comment:
the code instead is checking that the inserted element is a sibling of bbb, someone is not telling the truth here!